Style fixes on the previous merge request

jopedroliveira opened the pull request.

chenguoguo commented:
Thanks. I added some comments.  A high level suggestion: could we make this demo reading from raw audio file only, instead of making it for reading from both file and sound card?

The idea is that we already have demo.cc that reads from sound card, so we probably don't want to include it in demo2.cc

jopedroliveira commented:
@chenguoguo Yes, we can do that. 
I'll review the code under that perspective then I post a new merge.

jopedroliveira commented:
Ok, @chenguoguo 
Just reviewed the code and removed everything with the microphone audio acquisition and added a new feature. Now we can also read audio from a wav file.

jopedroliveira commented:
@chenguoguo meanwhile I noted the wav header of snowboy file had an error and fixed it. Also added a .RAW file. You can test the program with both snowboy.wav or snowboy.raw files

chenguoguo commented on examples/C++/demo2.cpp:
```diff
@@ -0,0 +1,145 @@
+    perror("WAV file has encoded data or it is WAVEFORMATEXTENSIBLE.");
+  }
+  if (wavhdr->fmtSize == 14){
+      wavhdr->bps = 16;
```
Please fix the indentation spaces here, should be 2.

chenguoguo commented on examples/C++/demo2.cpp:
```diff
@@ -0,0 +1,145 @@
+      wavhdr->bps = 16;
+  }
+  if (wavhdr->fmtSize >= 16){
+      fread(&wavhdr->bps, 2, 1, fi);
```
Same as the above, fix the indentation.

chenguoguo commented on examples/C++/demo2.cpp:
```diff
@@ -0,0 +1,145 @@
+
+
+int main(int argc, char * argv[]) {
+ char * usage = (char *) "C++ demo that shows how to use snowboy. \
```
Two spaces for indentation. Please fix the code below as well.

chenguoguo commented on examples/C++/demo2.cpp:
```diff
@@ -0,0 +1,145 @@
+
+ if(argc > 1){
+  filename = argv[1]; //input: file. else, input: mic
+ }
```
You already made sure argc == 2 above. So you don't need this if.

chenguoguo commented on examples/C++/demo2.cpp:
```diff
@@ -0,0 +1,145 @@
+ if(filename != NULL)
+  f = fopen(filename,"rb");
+  
+  if( f == NULL ){
```
Don't use space around `f == NULL`

jopedroliveira commented:
@chenguoguo how's the merge going?

jopedroliveira commented:
@chenguoguo I already addresed those issues 😄 
check [the last commit](https://github.com/Kitt-AI/snowboy/pull/291/commits/9bcda9dedf1140a5b4b684d07bef91726d6f9f21)

chenguoguo commented:
OK I'm going to merge it and fix those issues by myself. Some of the comments have not been addressed.
